compiler/executable/match_/planner/plan.rs

```diff
@@ -606,8 +612,9 @@ impl<'a> ConjunctionPlanBuilder<'a> {
         let cost = ordering
             .iter()
             .enumerate()
-            .map(|(i, idx)| self.graph.elements[idx].cost(&ordering[..i], &self.graph))
-            .fold(ElementCost::FREE, |acc, e| acc.chain(e));
+            // TODO: how do we pass the intersection vars back in to recompute the full cost?
+            .map(|(i, idx)| self.graph.elements[idx].cost(&ordering[..i], None, &self.graph))
+            .fold(ElementCost::FREE_BRANCH_1, |acc, e| acc.chain(e));
```

Comment:
We now pass the intersection variable into the `cost` function... this becomes a problem here since we want to re-compute the cost, but we no longer have access to the intersection variable!

Reply:
It should be inferrable from the ordering itself (viz. the next variable in the ordering should be the intersection).